### PR TITLE
Jake Henry node-express-mini

### PR DESCRIPTION
**Pull request by Blast3d**

### CODE REVIEW

**Comment by Blast3d** [comment]
This is Jacob henry, lol I have no way of contacting Lambda aside from gmail, hopefully someone sees this.

**Comment by enitchals** [comment]
Hey Jacob, just noticed this and gave Moises a heads-up 👍

**Comment by Blast3d** [comment]
ok awesome! thx

**Comment by enitchals** [comment]
Update: Word is they're updating Slack and you should be re-added shortly :)

**Comment by Blast3d** [comment]
..and piazza...? I was kicked from both, lol

**Comment by ghost** [comment]
Are you still in the repl?

**Comment by Blast3d** [comment]
yes

**Comment by enitchals** [comment]
I'm guessing that Slack and Piazza are linked. I'd recommend you follow up with Moises if you still don't have Piazza access after you're back in Slack.

**Comment by Blast3d** [comment]
ok thanks!

**Comment by designerexpert** [comment]
# Comments
* It seems you have a well executed express server. Implemented all the required endpoints. 👍 
* I think the only thing missing here is the stretch. Maybe try your hand at it?

File: server.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  db
+    .insert({
+      name,
+      bio
+    })
+    .then(response => {
+      res.json(response);
+    })
+    .catch(error => {
+      res.json(error);
```
Comment (nphillips78) [inline server.js:27]: Status code?

File: server.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+server.get('/api/users/:id', (req, res) => {
+  // pull id off of req.params;
+  // invoke proper db.method(id) passing it the id.
+  // handle the promise like above
+  const {
+    id
+  } = req.params;
+
```
Comment (nphillips78) [inline server.js:53]: Don't forget to set up an error message for if there's no user with that ID. The error in your catch is a server error, it doesn't cover the user not existing.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  db
+    .findById(id)
+    .then(users => {
+      res.json(users[0]);
```
Comment (nphillips78) [inline server.js:57]: Good

File: server.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    id
+  } = req.params;
+  const update = req.body;
+
+  db
+    .update(id, update)
+    .then(count => {
+      if (count > 0) {
+        db.findById(id).then(updatedUsers => {
+          res.status(200).json(updatedUsers[0]);
```
Comment (nphillips78) [inline server.js:102]: I like how you wrote this to check for the id before executing and putting the user error after.

**Review by nphillips78 — COMMENTED** [review]
Good job! Well executed server, sound logic. There's a couple of spots where you forgot a status code or forgot to code for the user error but overall great work!

**Pull request closed without merging**